### ISSUE TITLE
LockService::unlockSimple marked @NonIdempotent. It would be nice if it wasn't

### ISSUE THREAD

**Issue by mglazer** [palantir/atlasdb#1042]
Looking at the code for `LockServiceImpl::unlockSimple`, I can see why it is marked as such, because if you unlock something that somehow failed part way through, then the second time you call `unlockSimple`, it might do nothing at all.

However, it seems reasonable enough that if I call `unlockSimple` once, it fails (due to a network error, not application error), and then I call `unlockSimple` again, it should simply do nothing. 

Current behavior propagates all the way up the stack resulting in a state where I'm unsure of the best way to recover.

This ticket is the result of seeing a 'Connection Reset' error (a very common network error which seems to happen at alarming frequency on many networks) when performing `unlockSimple`, and that error being propagated all the way up as a user facing error.

**Comment by jboreiko** [comment]
This is related to #810.

We would like a sense of the impact @mglazer so we can decide how to prioritize this.  Happy to discuss off ticket if necessary.

**Comment by mglazer** [comment]
Not sure how related it is #810, but I'll assume it is.

Anyways, this was more of an annoyance than anything else. The way we handle failures of NonIdempotent operations is to fail hard, and kill the entire stack. Yes, this is horrible behavior, but it also prevents a really horrible class of data corruption issues. 

Occasional network blips happen though, and it's hard to discern between a network blip, and a valid lock server failure. In this case, because unlockSimple is so simple, I'll assume this was a network blip, which means LockServer never received the request, but that's purely an assumption, based off of data I can't gather anymore because it's on an inaccessible network.

So for stability purposes, it would be nice if any operations that can be made Idempotent, should be made idempotent.

Does that seem reasonable?

**Comment by jboreiko** [comment]
Yes, this does make sense.   Sorry for the slow response here, we haven't had time to prioritize this.  I still don't quite understand, is there error handling logic that is parsing the annotation or are you simply interested in seeing this made truly idempotent so that error handling can be changed elsewhere?
